### DISCUSSION TITLE
Official Deno Version Manager

### DISCUSSION THREAD

**Discussion by trymeouteh** [denoland/deno#26848] [Ideas]
To have a seperate CLI application that will allow you to install and manage different versions of Deno and be the official way to install Deno.

Perhaps by adding new commands in deno such as...

```
deno set 2.0.0
deno install 2.0.0
deno default 2.0.0
deno list
```

And to allow deno to detect if a file in the directory demands the current deno version is changed such as a `deno-version.json` file

**Comment by ry** [comment]
deno itself is its own version manager: `deno upgrade 1.46.0`

**Reply by trymeouteh** [reply]
Is there a way to download multiple versions at once and easily switch between each version? When I use `deno upgrade` it always downloads the version I switch to even if I used that version before.

**Reply by davesteinberg** [reply]
Another existing option is [asdf](https://asdf-vm.com/), which manages versions for a huge number of runtimes and tools, [deno](https://github.com/asdf-community/asdf-deno) included.

**Comment by mogwai-dev** [comment]
asdf seems to be only for OS X or Linux.  
On Windows, you need to use [dvm](https://github.com/justjavac/dvm).  
By the way, on my company PC WSL is not available, so I have to rely on dvm.  
It’s a bit annoying to install dvm separately, so I wish Deno had a built-in version manager.
